### PR TITLE
use pytest instead of nosetests

### PR DESCRIPTION
**Pull request by jreidinger**
experimental switch to run tests with pytest

### CODE REVIEW

File: dist/ci/testenv-tumbleweed/Dockerfile [hunk 1]
```diff
@@ -5,9 +5,9 @@ FROM opensuse/tumbleweed
 RUN zypper -n ar http://download.opensuse.org/repositories/openSUSE:/Tools/openSUSE_Factory/ openSUSE:Tools
 RUN zypper --gpg-auto-import-keys ref
 
-RUN zypper in -y osc python3-nose python3-httpretty python3-pyxdg python3-PyYAML \
+RUN zypper in -y osc python3-pytest python3-httpretty python3-pyxdg python3-PyYAML \
```
Comment (coolo) [inline dist/ci/testenv-tumbleweed/Dockerfile:8]: you better create another PR to add both. You can still remove nose in this.

Comment (jreidinger) [reply dist/ci/testenv-tumbleweed/Dockerfile:8]: well, it is still WIP. I plan to temporary just for testing install it here. Locally it failed after several tests for unavailable obs API, but it is probably caused by my local machine computation power.

Comment (coolo) [reply dist/ci/testenv-tumbleweed/Dockerfile:8]: that's my point: you can't test pytest without it being in the OBS container

Comment (jreidinger) [reply dist/ci/testenv-tumbleweed/Dockerfile:8]: well, I plan to install it with zypper for now as temporary solution for testing that all tests can pass.

Comment (jreidinger) [reply dist/ci/testenv-tumbleweed/Dockerfile:8]: trick we use often in yast where we have shared base image for yast and if additional library is needed for one specific module we install it with zypper as part of run to not grow too much generic image.

**Comment by jreidinger** [comment]
OK, so it runs faster...but failing. I will need to debug what exactly and why failed.

File: dist/ci/docker-compose-test.sh [hunk 2]
```diff
@@ -13,13 +16,8 @@ until curl http://api:3000/about 2>/dev/null ; do
...
 cd /code
-for file in tests/*_tests.py; do
-  if ! test -f /code/.without-coverage; then
-    COVER_ARGS="--with-coverage --cover-xml --cover-package=. --cover-inclusive"
-  fi
-  echo "running tests from $file..."
-  run_as_tester nosetests $COVER_ARGS -c .noserc -s $file
-done
+# TODO: Review bot test failed without log-level set to debug
+run_as_tester pytest --log-level=DEBUG --cov=. tests/*_tests.py
```
Comment (jreidinger) [inline dist/ci/docker-compose-test.sh:20]: this log level is strange...maybe nosetests has it by default and it can be reason why pytest failed in that review bot test.

File: dist/ci/docker-compose-test.sh [hunk 3]
```diff
@@ -13,13 +16,13 @@ until curl http://api:3000/about 2>/dev/null ; do
...
 cd /code
-for file in tests/*_tests.py; do
-  if ! test -f /code/.without-coverage; then
-    COVER_ARGS="--with-coverage --cover-xml --cover-package=. --cover-inclusive"
-  fi
-  echo "running tests from $file..."
-  run_as_tester nosetests $COVER_ARGS -c .noserc -s $file
+for i in tests/*_tests.py; do
+  # TODO: Review bot test failed without log-level set to debug
+  # FIXME: JR: for not yet known reason running all tests together failed
```
Comment (coolo) [inline dist/ci/docker-compose-test.sh:21]: there is global state kept in osc library and in memoize

**Comment by coolo** [comment]
At least the speed is on par now :smile:

**Comment by jreidinger** [comment]
@coolo yeah, right. Thanks for explain why running all tests together does not work. Maybe reseting that memoize between tests can help as now annoying part is that it fail in first failed test and you do not see all failures at once.

**Comment by coolo** [comment]
I tried for quite a while - but memoize is not the only global state. So I gave up in favor of having test coverage at all :(

File: dist/ci/docker-compose-test.sh [hunk 4]
```diff
@@ -13,12 +13,10 @@ until curl http://api:3000/about 2>/dev/null ; do
 done
 
 cd /code
-for file in tests/*_tests.py; do
-  if ! test -f /code/.without-coverage; then
-    COVER_ARGS="--with-coverage --cover-xml --cover-package=. --cover-inclusive"
-  fi
-  echo "running tests from $file..."
-  run_as_tester nosetests $COVER_ARGS -c .noserc -s $file
+for i in tests/*_tests.py; do
```
Comment (coolo) [inline dist/ci/docker-compose-test.sh:16]: hmm, why did you rename my variable `file` now `i`? 

Why did you make it imposible to run without coverage now? it's a waste of time while developing locally

**Review by coolo — CHANGES_REQUESTED** [review]

File: dist/ci/docker-compose-test.sh (hunk 4 above)
Comment (jreidinger) [reply dist/ci/docker-compose-test.sh:16]: ok, will revert it back.

**Review by coolo — APPROVED** [review]
